crates/unleash-edge-auth/src/token_validator.rs

```diff
@@ -46,6 +47,46 @@ impl TokenRegister for TokenValidator {
     }
 }
 
+pub fn create_revalidation_task(
+    validator: &Arc<TokenValidator>,
+    validation_interval_seconds: u64,
+) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+    let validator_clone = validator.clone();
+    Box::pin(async move {
+        let sleep_duration = tokio::time::Duration::from_secs(validation_interval_seconds);
+        loop {
+            tokio::select! {
+                _ = tokio::time::sleep(sleep_duration) => {
+                    let _ = validator_clone.revalidate_known_tokens().await;
+                }
+            }
+        }
+    })
+}
+
+pub fn create_revalidation_of_startup_tokens_task(
```

Comment:
Same thing, schedule_revalidation_of_startup_tokens is now this + Box::pin(async move{})